chaosazure/vmss/fetcher.py

```diff
@@ -6,18 +6,33 @@
 from logzero import logger
 
 from chaosazure import init_client
-from chaosazure.rgraph.resource_graph import fetch_resources
+from chaosazure.common.resources.graph import fetch_resources
 from chaosazure.vmss.constants import RES_TYPE_VMSS
 
 
+def fetch_instances(scale_set, instance_criteria,
+                    configuration, secrets) -> List[Dict[str, Any]]:
+    if not instance_criteria:
+        instance = choose_vmss_instance_at_random(
+            scale_set, configuration, secrets)
+        result = [instance]
+
+    else:
+        result = choose_vmss_instance(
+            scale_set, configuration, instance_criteria, secrets)
+
+    return result
+
+
 def choose_vmss_instance_at_random(vmss_choice, configuration, secrets):
     vmss_instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
     if not vmss_instances:
-        logger.warning("No virtual machine scale set instances found")
-        raise FailedActivity("No virtual machine scale set instances found")
+        msg = "No VMSS instances found"
+        logger.warning(msg)
```

Comment:
Is it warning or error? In next line we are failing the activity. I think that failed activity will log as error. So we will have 2 logs one warning and one error with the same message. Probably this one is redundant

Reply:
Solved